### PR TITLE
P2P review updates

### PR DESCRIPTION
**Pull request by vimukthi-git**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: <Github link>

### CODE REVIEW

File: p2p/server.go [hunk 1]
```diff
@@ -53,8 +53,8 @@ type messenger interface {
 	sendMessage(ctx context.Context, p peer.ID, pmes *pb.P2PEnvelope, protoc protocol.ID) (*pb.P2PEnvelope, error)
 }
 
-// p2pServer implements api.Server
-type p2pServer struct {
+// centPeer implements api.Server
+type centPeer struct {
```
[the remark this answers is not in the document]
Comment (vimukthi-git) [reply p2p/server.go]: I also did this at first, this cant be done as we import a libp2p lib as `peer` everywhere, I would keep it as I don't have any other name in my head that I could use.
cc @lucasvo

[the remark this answers is not in the document]
Comment (mikiquantum) [reply p2p/server.go]: So we cannot keep it as `p2pServer` then? It is technically one of the servers (api and p2p), a peer has different meaning IMO.

[the remark this answers is not in the document]
Comment (lucasvo) [reply p2p/server.go]: I actually requested that change @mikiquantum the "Server" struct implements both handling of incoming requests and sending outgoing requests. I suggested to call it `peer` or `node` for that reason. I was confused when encountering a p2pServer object when it was used as a "client".

[the remark this answers is not in the document]
Comment (lucasvo) [reply p2p/server.go]: In my comment I actually proposed the name node, not peer but either one is better than server IMO. See the discussion here: https://github.com/centrifuge/go-centrifuge/pull/597/files#diff-6af2086d270826b158fbdffe1affe9bbR36

[the remark this answers is not in the document]
Comment (mikiquantum) [reply p2p/server.go]: That server/client logic was due to gRPC. Now that we do not have it we can split that logic up into two different client/server structs. For the sake of advancing now I am fine to keep it as `peer` but that is confusing as well.

**Pull request merged**